Debug handler has incorrect borrow suggestion

Using `axum` and `axum-macros` upstream version (git main branch).

I got this error when using `#[debug_handler]` like so:
```rs
#[debug_handler]
async fn insert_multirow_once(DbConn(conn): DbConn) -> Result<impl IntoResponse, StatusCode> {
...
}
```

```
help: consider borrowing here
   |
34 | &async fn insert_multirow_once(DbConn(conn): DbConn) -> Result<impl IntoResponse, StatusCode> {
   | +
```

It seems to take from the next line, because this happens when a comment is added between the attribute and the function definition:
```
help: consider borrowing here
   |
34 | &/// A comment after the attribute
   | +
```

Full example [here](https://github.com/allan2/axum-postgres-dynamic-params/blob/main/src/main.rs)

It is generally best to read compile errors starting from the top. There you'll see

```
error[E0277]: the trait bound `Pool<PostgresConnectionManager<NoTls>>: FromRef<()>` is not satisfied
  --> src/main.rs:35:45
   |
35 | async fn insert_multirow_once(DbConn(conn): DbConn) -> Result<impl IntoResponse, StatusCode> {
   |                                             ^^^^^^ the trait `FromRef<()>` is not implemented for `Pool<PostgresConnectionManager<NoTls>>`
   |
   = help: the following other types implement trait `FromRequest<S, B, M>`:
             (T1, T2)
             (T1, T2, T3)
             (T1, T2, T3, T4)
             (T1, T2, T3, T4, T5)
             (T1, T2, T3, T4, T5, T6)
             (T1, T2, T3, T4, T5, T6, T7)
             (T1, T2, T3, T4, T5, T6, T7, T8)
             (T1, T2, T3, T4, T5, T6, T7, T8, T9)
           and 19 others
note: required for `DbConn` to implement `FromRequestParts<()>`
```

and a bit further down

```
error: future cannot be sent between threads safely
  --> src/main.rs:35:1
   |
35 | async fn insert_multirow_once(DbConn(conn): DbConn) -> Result<impl IntoResponse, StatusCode> {
   | ^^^^^ future returned by `insert_multirow_once` is not `Send`
   |
   = note: the trait bound `Unique<dyn ToSql + Sync>: Send` is not satisfied
```

Those are the important errors.

I don't believe there is anything we can do about the borrow suggestion. Thats probably just rustc being a little silly 😅

--- a/axum-macros/src/debug_handler.rs
+++ b/axum-macros/src/debug_handler.rs
@@ -373,7 +373,7 @@ fn check_future_send(item_fn: &ItemFn) -> TokenStream {
         };
     }
 
-    let span = item_fn.span();
+    let span = item_fn.sig.ident.span();
 
     let handler_name = &item_fn.sig.ident;
 
@@ -383,28 +383,28 @@ fn check_future_send(item_fn: &ItemFn) -> TokenStream {
 
     let name = format_ident!("__axum_macros_check_{}_future", item_fn.sig.ident);
 
+    let do_check = quote! {
+        fn check<T>(_: T)
+            where T: ::std::future::Future + Send
+        {}
+        check(future);
+    };
+
     if let Some(receiver) = self_receiver(item_fn) {
-        quote_spanned! {span=>
+        quote! {
             #[allow(warnings)]
             fn #name() {
                 let future = #receiver #handler_name(#(#args),*);
-                fn check<T>(_: T)
-                    where T: ::std::future::Future + Send
-                {}
-                check(future);
+                #do_check
             }
         }
     } else {
-        quote_spanned! {span=>
+        quote! {
             #[allow(warnings)]
             fn #name() {
                 #item_fn
-
                 let future = #handler_name(#(#args),*);
-                fn check<T>(_: T)
-                    where T: ::std::future::Future + Send
-                {}
-                check(future);
+                #do_check
             }
         }
     }

--- a/axum-macros/tests/debug_handler/fail/not_send.stderr
+++ b/axum-macros/tests/debug_handler/fail/not_send.stderr
@@ -1,11 +1,8 @@
 error: future cannot be sent between threads safely
- --> tests/debug_handler/fail/not_send.rs:4:1
+ --> tests/debug_handler/fail/not_send.rs:3:1
   |
-4 | / async fn handler() {
-5 | |     let rc = std::rc::Rc::new(());
-6 | |     async {}.await;
-7 | | }
-  | |_^ future returned by `handler` is not `Send`
+3 | #[debug_handler]
+  | ^^^^^^^^^^^^^^^^ future returned by `handler` is not `Send`
   |
   = help: within `impl Future<Output = ()>`, the trait `Send` is not implemented for `Rc<()>`
 note: future is not `Send` as this value is used across an await
@@ -18,10 +15,8 @@ note: future is not `Send` as this value is used across an await
 7 | }
   | - `rc` is later dropped here
 note: required by a bound in `check`
- --> tests/debug_handler/fail/not_send.rs:4:1
+ --> tests/debug_handler/fail/not_send.rs:3:1
   |
-4 | / async fn handler() {
-5 | |     let rc = std::rc::Rc::new(());
-6 | |     async {}.await;
-7 | | }
-  | |_^ required by this bound in `check`
+3 | #[debug_handler]
+  | ^^^^^^^^^^^^^^^^ required by this bound in `check`
+  = note: this error originates in the attribute macro `debug_handler` (in Nightly builds, run with -Z macro-backtrace for more info)

--- a/axum-macros/tests/debug_handler/fail/wrong_return_type.stderr
+++ b/axum-macros/tests/debug_handler/fail/wrong_return_type.stderr
@@ -18,4 +18,4 @@ note: required by a bound in `__axum_macros_check_handler_into_response::{closur
  --> tests/debug_handler/fail/wrong_return_type.rs:4:23
   |
 4 | async fn handler() -> bool {
-  |                       ^^^^ required by this bound in `check`
+  |                       ^^^^ required by this bound in `__axum_macros_check_handler_into_response::{closure#0}::check`

--- a/axum-macros/tests/debug_handler/fail/wrong_return_type.stderr
+++ b/axum-macros/tests/debug_handler/fail/wrong_return_type.stderr
@@ -13,9 +13,9 @@ error[E0277]: the trait bound `bool: IntoResponse` is not satisfied
             (Response<()>, T1, R)
             (Response<()>, T1, T2, R)
             (Response<()>, T1, T2, T3, R)
-          and 124 others
+          and $N others
 note: required by a bound in `__axum_macros_check_handler_into_response::{closure#0}::check`
  --> tests/debug_handler/fail/wrong_return_type.rs:4:23
   |
 4 | async fn handler() -> bool {
-  |                       ^^^^ required by this bound in `__axum_macros_check_handler_into_response::{closure#0}::check`
+  |                       ^^^^ required by this bound in `check`

--- a/axum-macros/tests/from_request/fail/generic_without_via.stderr
+++ b/axum-macros/tests/from_request/fail/generic_without_via.stderr
@@ -5,18 +5,18 @@ error: #[derive(FromRequest)] only supports generics when used with #[from_reque
   |                  ^
 
 error[E0277]: the trait bound `fn(Extractor<()>) -> impl Future<Output = ()> {foo}: Handler<_, _, _>` is not satisfied
-   --> tests/from_request/fail/generic_without_via.rs:10:46
-    |
-10  |     Router::<(), Body>::new().route("/", get(foo));
-    |                                          --- ^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(Extractor<()>) -> impl Future<Output = ()> {foo}`
-    |                                          |
-    |                                          required by a bound introduced by this call
-    |
-    = note: Consider using `#[axum::debug_handler]` to improve the error message
-    = help: the trait `Handler<T, S, B2>` is implemented for `Layered<L, H, T, S, B, B2>`
+  --> tests/from_request/fail/generic_without_via.rs:10:46
+   |
+10 |     Router::<(), Body>::new().route("/", get(foo));
+   |                                          --- ^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(Extractor<()>) -> impl Future<Output = ()> {foo}`
+   |                                          |
+   |                                          required by a bound introduced by this call
+   |
+   = note: Consider using `#[axum::debug_handler]` to improve the error message
+   = help: the trait `Handler<T, S, B2>` is implemented for `Layered<L, H, T, S, B, B2>`
 note: required by a bound in `axum::routing::get`
-   --> $WORKSPACE/axum/src/routing/method_routing.rs
-    |
-    | top_level_handler_fn!(get, GET);
-    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `axum::routing::get`
-    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)
+  --> $WORKSPACE/axum/src/routing/method_routing.rs
+   |
+   | top_level_handler_fn!(get, GET);
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `axum::routing::get`
+   = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

--- a/axum-macros/tests/from_request/fail/generic_without_via_rejection.stderr
+++ b/axum-macros/tests/from_request/fail/generic_without_via_rejection.stderr
@@ -5,18 +5,18 @@ error: #[derive(FromRequest)] only supports generics when used with #[from_reque
   |                  ^
 
 error[E0277]: the trait bound `fn(Extractor<()>) -> impl Future<Output = ()> {foo}: Handler<_, _, _>` is not satisfied
-   --> tests/from_request/fail/generic_without_via_rejection.rs:11:46
-    |
-11  |     Router::<(), Body>::new().route("/", get(foo));
-    |                                          --- ^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(Extractor<()>) -> impl Future<Output = ()> {foo}`
-    |                                          |
-    |                                          required by a bound introduced by this call
-    |
-    = note: Consider using `#[axum::debug_handler]` to improve the error message
-    = help: the trait `Handler<T, S, B2>` is implemented for `Layered<L, H, T, S, B, B2>`
+  --> tests/from_request/fail/generic_without_via_rejection.rs:11:46
+   |
+11 |     Router::<(), Body>::new().route("/", get(foo));
+   |                                          --- ^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(Extractor<()>) -> impl Future<Output = ()> {foo}`
+   |                                          |
+   |                                          required by a bound introduced by this call
+   |
+   = note: Consider using `#[axum::debug_handler]` to improve the error message
+   = help: the trait `Handler<T, S, B2>` is implemented for `Layered<L, H, T, S, B, B2>`
 note: required by a bound in `axum::routing::get`
-   --> $WORKSPACE/axum/src/routing/method_routing.rs
-    |
-    | top_level_handler_fn!(get, GET);
-    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `axum::routing::get`
-    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)
+  --> $WORKSPACE/axum/src/routing/method_routing.rs
+   |
+   | top_level_handler_fn!(get, GET);
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `axum::routing::get`
+   = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

--- a/axum-macros/tests/from_request/fail/override_rejection_on_enum_without_via.stderr
+++ b/axum-macros/tests/from_request/fail/override_rejection_on_enum_without_via.stderr
@@ -5,35 +5,35 @@ error: cannot use `rejection` without `via`
    |                ^^^^^^^^^
 
 error[E0277]: the trait bound `fn(MyExtractor) -> impl Future<Output = ()> {handler}: Handler<_, _, _>` is not satisfied
-   --> tests/from_request/fail/override_rejection_on_enum_without_via.rs:10:50
-    |
-10  |     let _: Router = Router::new().route("/", get(handler).post(handler_result));
-    |                                              --- ^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(MyExtractor) -> impl Future<Output = ()> {handler}`
-    |                                              |
-    |                                              required by a bound introduced by this call
-    |
-    = note: Consider using `#[axum::debug_handler]` to improve the error message
-    = help: the trait `Handler<T, S, B2>` is implemented for `Layered<L, H, T, S, B, B2>`
+  --> tests/from_request/fail/override_rejection_on_enum_without_via.rs:10:50
+   |
+10 |     let _: Router = Router::new().route("/", get(handler).post(handler_result));
+   |                                              --- ^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(MyExtractor) -> impl Future<Output = ()> {handler}`
+   |                                              |
+   |                                              required by a bound introduced by this call
+   |
+   = note: Consider using `#[axum::debug_handler]` to improve the error message
+   = help: the trait `Handler<T, S, B2>` is implemented for `Layered<L, H, T, S, B, B2>`
 note: required by a bound in `axum::routing::get`
-   --> $WORKSPACE/axum/src/routing/method_routing.rs
-    |
-    | top_level_handler_fn!(get, GET);
-    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `axum::routing::get`
-    = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)
+  --> $WORKSPACE/axum/src/routing/method_routing.rs
+   |
+   | top_level_handler_fn!(get, GET);
+   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `axum::routing::get`
+   = note: this error originates in the macro `top_level_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)
 
 error[E0277]: the trait bound `fn(Result<MyExtractor, MyRejection>) -> impl Future<Output = ()> {handler_result}: Handler<_, _, _>` is not satisfied
-   --> tests/from_request/fail/override_rejection_on_enum_without_via.rs:10:64
-    |
-10  |     let _: Router = Router::new().route("/", get(handler).post(handler_result));
-    |                                                           ---- ^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(Result<MyExtractor, MyRejection>) -> impl Future<Output = ()> {handler_result}`
-    |                                                           |
-    |                                                           required by a bound introduced by this call
-    |
-    = note: Consider using `#[axum::debug_handler]` to improve the error message
-    = help: the trait `Handler<T, S, B2>` is implemented for `Layered<L, H, T, S, B, B2>`
+  --> tests/from_request/fail/override_rejection_on_enum_without_via.rs:10:64
+   |
+10 |     let _: Router = Router::new().route("/", get(handler).post(handler_result));
+   |                                                           ---- ^^^^^^^^^^^^^^ the trait `Handler<_, _, _>` is not implemented for fn item `fn(Result<MyExtractor, MyRejection>) -> impl Future<Output = ()> {handler_result}`
+   |                                                           |
+   |                                                           required by a bound introduced by this call
+   |
+   = note: Consider using `#[axum::debug_handler]` to improve the error message
+   = help: the trait `Handler<T, S, B2>` is implemented for `Layered<L, H, T, S, B, B2>`
 note: required by a bound in `MethodRouter::<S, B>::post`
-   --> $WORKSPACE/axum/src/routing/method_routing.rs
-    |
-    |     chained_handler_fn!(post, POST);
-    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MethodRouter::<S, B>::post`
-    = note: this error originates in the macro `chained_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)
+  --> $WORKSPACE/axum/src/routing/method_routing.rs
+   |
+   |     chained_handler_fn!(post, POST);
+   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ required by this bound in `MethodRouter::<S, B>::post`
+   = note: this error originates in the macro `chained_handler_fn` (in Nightly builds, run with -Z macro-backtrace for more info)

--- a/axum-macros/tests/typed_path/fail/not_deserialize.stderr
+++ b/axum-macros/tests/typed_path/fail/not_deserialize.stderr
@@ -13,7 +13,7 @@ error[E0277]: the trait bound `for<'de> MyPath: serde::de::Deserialize<'de>` is 
             (T0, T1)
             (T0, T1, T2)
             (T0, T1, T2, T3)
-          and 129 others
+          and $N others
   = note: required for `MyPath` to implement `serde::de::DeserializeOwned`
   = note: required for `axum::extract::Path<MyPath>` to implement `FromRequestParts<S>`
   = note: this error originates in the derive macro `TypedPath` (in Nightly builds, run with -Z macro-backtrace for more info)

